### PR TITLE
JIT: move indirect call transformations to a new source file

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Follow-up from a review comment in #21270.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@sandreenko PTAL
cc @dotnet/jit-contrib 

No source code changes; a few minor comment edits.

**Comment by AndyAyersMS** [comment]
Perf test failures are known issues. Fix is #21412.

File: src/jit/indirectcalltransformer.cpp [hunk 1]
```diff
@@ -0,0 +1,839 @@
...
+// or load hidden argument, fix the pointer and make a call with the fixed pointer and
+// the instantiation argument.
+//
+// before:
+//   current block
+//   {
+//     previous statements
+//     transforming statement
+//     {
+//       call with GTF_CALL_M_FAT_POINTER_CHECK flag set in function ptr
```
Comment (CarolEidt) [inline src/jit/indirectcalltransformer.cpp]: I think that this should say `GTF_CALL_M_FAT_POINTER_CHECK flag set on the call node.`

Comment (CarolEidt) [reply src/jit/indirectcalltransformer.cpp]: What I meant was "FAT_POINTER_MASK"

**Comment by AndyAyersMS** [comment]
@CarolEidt I have another edit in progress up for this file, so will fix the issues you note above there.

**Pull request merged**